From dd224a3b52774c387f1410aaaddf202f1ee88cec Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Tue, 7 Aug 2018 17:17:29 +0200
Subject: [PATCH 3/7] debuggerd: Resolve tombstoned missing O_CREAT mode

 debuggerd/tombstoned/tombstoned.cpp:115:92:
  error: 'openat' has superfluous mode bits; missing O_CREAT?
    unique_fd result(openat(dir_fd_, ".",
    O_WRONLY | O_APPEND | O_TMPFILE | O_CLOEXEC, 0640));

Change-Id: Ifcc8f41063e36722c5c2a0a5d0bf64ebe2b1f910
---
 debuggerd/tombstoned/tombstoned.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debuggerd/tombstoned/tombstoned.cpp b/debuggerd/tombstoned/tombstoned.cpp
index 15ae40624..e5d62b1b8 100644
--- a/debuggerd/tombstoned/tombstoned.cpp
+++ b/debuggerd/tombstoned/tombstoned.cpp
@@ -112,7 +112,7 @@ class CrashQueue {
 
   std::pair<std::string, unique_fd> get_output() {
     std::string path;
-    unique_fd result(openat(dir_fd_, ".", O_WRONLY | O_APPEND | O_TMPFILE | O_CLOEXEC, 0640));
+    unique_fd result(openat(dir_fd_, ".", O_CREAT | O_WRONLY | O_APPEND | O_TMPFILE | O_CLOEXEC, 0640));
     if (result == -1) {
       // We might not have O_TMPFILE. Try creating with an arbitrary filename instead.
       static size_t counter = 0;
-- 
2.17.1

